metrics/thresholds_test.go

```diff
@@ -334,6 +337,219 @@ func TestThresholdsParse(t *testing.T) {
 	})
 }
 
+func TestThresholdsValidate(t *testing.T) {
+	t.Parallel()
+
+	t.Run("validating thresholds applied to a non-existing metric fails", func(t *testing.T) {
+		t.Parallel()
+
+		testRegistry := NewRegistry()
+
+		// Prepare a Thresholds collection containing syntaxically
+		// correct thresholds
+		ts := Thresholds{
+			Thresholds: []*Threshold{
+				newThreshold("rate<1", false, types.NullDuration{}),
+			},
+		}
+
+		parseErr := ts.Parse()
+		require.NoError(t, parseErr)
+
+		var wantErr errext.HasExitCode
+
+		// Collect the result of the parsing operation
+		gotErr := ts.Validate("non-existing", testRegistry)
+
+		assert.Error(t, gotErr, "Validate should fail when applied to a non-existing metric")
+		assert.ErrorIs(t, gotErr, ErrInvalidThreshold, "Validate error chain should contain an ErrInvalidThreshold error")
```

Comment:
```suggestion
		assert.ErrorIs(t, gotErr, ErrInvalidThreshold)
```
This form seems already clear to me